selfdrive/car/body/carcontroller.py

```diff
@@ -66,25 +87,68 @@ def update(self, CC, CS):
       torque_diff = self.turn_pid.update(turn_error, freeze_integrator=freeze_integrator)
 
       # Combine 2 PIDs outputs
-      torque_r = torque + torque_diff
       torque_l = torque - torque_diff
+      torque_r = torque + torque_diff
 
       # Torque rate limits
-      self.torque_r_filtered = np.clip(self.deadband_filter(torque_r, 10),
-                                       self.torque_r_filtered - MAX_TORQUE_RATE,
-                                       self.torque_r_filtered + MAX_TORQUE_RATE)
       self.torque_l_filtered = np.clip(self.deadband_filter(torque_l, 10),
                                        self.torque_l_filtered - MAX_TORQUE_RATE,
                                        self.torque_l_filtered + MAX_TORQUE_RATE)
-      torque_r = int(np.clip(self.torque_r_filtered, -MAX_TORQUE, MAX_TORQUE))
+      self.torque_r_filtered = np.clip(self.deadband_filter(torque_r, 10),
+                                       self.torque_r_filtered - MAX_TORQUE_RATE,
+                                       self.torque_r_filtered + MAX_TORQUE_RATE)
+
       torque_l = int(np.clip(self.torque_l_filtered, -MAX_TORQUE, MAX_TORQUE))
+      torque_r = int(np.clip(self.torque_r_filtered, -MAX_TORQUE, MAX_TORQUE))
+
+    # Knee angle controls
+    if self.CP.carFingerprint == CAR.BODY_KNEE and CC.enabled:
+      knee_angle_desired_left = CC.actuators.gas
+      knee_angle_desired_right = CC.actuators.brake
+
+      knee_angle_measured_left = CS.knee_angle_l
+      knee_angle_measured_right = CS.knee_angle_r
```

Comment:
these should be in a cereal struct

Reply:
in car state? say as SensorAngles struct?